instrumentation/servlet/src/main/java/brave/servlet/HttpServletResponseWrapper.java

```diff
@@ -31,29 +30,37 @@
 // Public for use in sparkjava or other frameworks that re-use servlet types
 public class HttpServletResponseWrapper extends HttpServerResponse { // not final for inner subtype
   /**
-   * Looks for the {@link HttpServletRequest#setAttribute(String, Object) request attributes}
-   * "http.route" and "error" to customize the result.
-   *
    * @param caught an exception caught serving the request.
    * @since 5.10
    */
-  public static HttpServerResponse create(@Nullable HttpServletRequest req,
-    HttpServletResponse res, @Nullable Throwable caught) {
-    if (req == null) return new HttpServletResponseWrapper(res, caught);
-    return new WithRequestProperties(req, res, caught);
+  public static HttpServerResponse create(@Nullable HttpServletRequest request,
+    HttpServletResponse response, @Nullable Throwable caught) {
+    return new HttpServletResponseWrapper(request, response, caught);
   }
 
-  final HttpServletResponse delegate;
+  @Nullable final HttpServletRequestWrapper request;
+  final HttpServletResponse response;
   @Nullable final Throwable caught;
 
-  HttpServletResponseWrapper(HttpServletResponse delegate, @Nullable Throwable caught) {
-    if (delegate == null) throw new NullPointerException("delegate == null");
-    this.delegate = delegate;
+  HttpServletResponseWrapper(@Nullable HttpServletRequest request, HttpServletResponse response,
+    @Nullable Throwable caught) {
+    if (response == null) throw new NullPointerException("response == null");
+    this.request = request != null ? new HttpServletRequestWrapper(request) : null;
+    this.response = response;
     this.caught = caught;
   }
 
+  @Override public HttpRequest request() {
+    return request;
+  }
+
+  @Override public Throwable error() {
+    if (caught != null || request == null) return caught;
```

Comment:
I had to do a doubletake here, it's an extra line but this could help readers

```java
if (caught != null) return caught;
if (request == null) return null;
return request.maybeError();
```